test/scf_compare.jl

```diff
@@ -1,9 +1,22 @@
 using Test
 using DFTK
 import DFTK: Applyχ0Model, select_occupied_orbitals
+using NLsolve
 
 include("testcases.jl")
 
+# Create a NLSolve-based SCF solver, by default using an Anderson-accelerated
+# fixed-point scheme, keeping `m` steps for Anderson acceleration. See the
+# NLSolve documentation for details about the other parameters and methods.
+function scf_nlsolve_solver(m=10, method=:anderson; kwargs...)
+    function fp_solver(f, x0, max_iter; tol=1e-6)
+        res = nlsolve(x -> f(x) - x, x0; method, m, xtol=tol,
+                      ftol=0.0, show_trace=false, iterations=max_iter, kwargs...)
+        (; fixpoint=res.zero, converged=converged(res))
+    end
+    fp_solver
+end
+
```

Comment:
I think if we really want to dump NLSolve then there is no need to reintroduce it just for the tests.